wherehows-web/app/components/dataset-compliance.js

```diff
@@ -802,18 +800,32 @@ export default Component.extend({
 
     /**
      * Receives the json representation for compliance and applies each key to the policy
-     * @param {String} textString string representation for the JSON file
+     * @param {string} textString string representation for the JSON file
      */
     onComplianceJsonUpload(textString) {
-      const policy = JSON.parse(textString);
+      let policy;
+      try {
+        policy = JSON.parse(textString);
+      } catch (e) {
+        get(this, 'notifications').notify('error', {
+          content: invalidPolicyData
+        });
+      }
+
       if (isPolicyExpectedShape(policy)) {
-        set(this, 'complianceInfo', policy);
+        setProperties(this, {
+          'complianceInfo.complianceEntities': policy.complianceEntities,
+          'complianceInfo.datasetClassification': policy.datasetClassification
+        });
 
-        // If all is good, then we can saveCompliance so user does not have to manually click
-        return this.actions.saveCompliance();
+        get(this, 'notifications').notify('info', {
```

Comment:
No return? it will show an error after this.

Reply:
ahah, this is what happens when you code on the Caltrain. good catch!